pypeit/scripts/chk_noise_1dspec.py

```diff
@@ -0,0 +1,224 @@
+        parser = super().get_parser(description='Examine the noise in a PypeIt spectrum',
+                                    width=width)
+        parser.add_argument('files', type = str, nargs='*', help = 'PypeIt spec1d file(s)')
+        parser.add_argument('--fileformat', default='spec1d', type=str, help='Is this coadd1d or spec1d?')
+        parser.add_argument('--extraction', default='opt', type=str, help='If spec1d, which extraction? opt or box [default: opt]')
+        parser.add_argument('--ploterr', default=False, help='Plot noise spectrum',action='store_true')
+        parser.add_argument('--step', default=False, help='Use `steps-mid` as linestyle',action='store_true')
+        parser.add_argument('--z', default=None, type=float, nargs='*', help='Object redshift')
+        parser.add_argument('--maskdef_objname', default=None, type=str, help='MASKDEF_OBJNAME of the target that you want to plot')
+        parser.add_argument('--pypeit_name', default=None, type=str, help='PypeIt name of the target that you want to plot')
+        parser.add_argument('--wavemin', default=None, type=float, help='Wavelength min. This is for selecting a region of the spectrum to analyze.')
+        parser.add_argument('--wavemax', default=None, type=float, help='Wavelength max.This is for selecting a region of the spectrum to analyze.')
+        parser.add_argument('--plot_or_save', default='plot', type=str, help='Do you want to save to disk or open a plot in a mpl window. If you choose save, a folder called spec1d*_noisecheck will be created and all the relevant plot will be placed there.')
+        args = parser.parse_args()
+
+        return parser
+
+    @staticmethod
+    def main(args):
+
+        maskdef_objname = None
+        # Load em
+        line_names, line_wav = utils.list_of_spectral_lines()
+            
+        files=np.array(args.files)
+
+        if args.z is not None:
+            zs = np.array(args.z) 
+
+        # Loop on the files
+        for i in range(files.size):    
+            # reinitialize lines wave
+            line_wav_z=line_wav.copy()
+
+            file = files[i]
+
+            # Deal with redshifts
+            if args.z is not None:
+                z = zs[i] if zs.size == files.size else zs[0]
+                line_wav_z *= (1+z)   #redshift linelist
+            else:
+                z = None
+
+            # Save?
+            if args.fileformat == 'spec1d':
+                if args.plot_or_save == 'save':
+                    folder = '{}_noisecheck'.format(file.split('.fits')[0])
+                    if not os.path.exists(folder): os.makedirs(folder)
+
+            # I/O
+            if args.fileformat == 'coadd1d':
+                coaddspec = OneSpec.from_file(file)
+                lbda = coaddspec.wave
+                flux = coaddspec.flux
+                err = coaddspec.sig
+                mask = coaddspec.mask
+            else:
+                specObjs = specobjs.SpecObjs.from_fitsfile(file)
```

Comment:
I don't expect this to be done for this PR, but I'm going to throw this idea out there.  It would be nice if the plotting routine was accessible to or a member of `SpecObjs` or `OneSpec`, or preferrably a parent/base class of both.  That way we could have something like:
```
specObjs = specobjs.SpecObjs.from_fitsfile(file)
specObjs.chk_noise()
```